org.adempiere.base/src/org/compiere/model/MAssetAddition.java

```diff
@@ -433,12 +433,9 @@ public void setSourceAmt(BigDecimal amt)
 	 */
 	public void setIsApproved()
 	{
-		if(!isProcessed())
+		if(!isProcessed() && MRole.getDefault().isCanApproveOwnDoc())
 		{
-			String str = Env.getContext(getCtx(), "#IsCanApproveOwnDoc");

```

Comment:
Perhaps we should add this variable instead, could be useful ?

Reply:
I think so, too. I may be useful in other circumstances, too.

Reply:
Done, implemented in last commit